### PR TITLE
Msal node error docs

### PR DESCRIPTION
**Pull request by mpminayo**
Documentation on errors thrown  by msal node.

### CODE REVIEW

File: lib/msal-node/docs/errors.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
+# Errors
+
+***
+
+**[ClientAuthErrors](#Clientautherrors)**
+
+1.[endpoints_resolution_error](#endpoints_resolution_error)
+
+2.[openid_config_error](#openid_config_error)
```
Comment (derisen) [inline lib/msal-node/docs/errors.md:9]: `#openid_config_error` link is busted

File: lib/msal-node/docs/errors.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+## ClientAuthErrors
+
+### Endpoints_resolution_error
+
+**Error Message**: Error: could not resolve endpoints. Please check network and try again.
+
+This error is thrown under the following scenarios:
+
+1. When a wrong authority endpoint is provided. Details: ClientAuthError: `openid_config_error`: Could not retrieve endpoints. Check your authority and verify the .well-known/openid-configuration endpoint returns the required endpoints.
```
Comment (derisen) [inline lib/msal-node/docs/errors.md:33]: > Details: ClientAuthError: `openid_config_error`: Could not retrieve endpoints. Check your authority and verify the .well-known/openid-configuration endpoint returns the required endpoints.

Maybe highlight this as error message to make it clear?

File: lib/msal-node/docs/errors.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+```
+
+
+## ClientConfigurationErrors
+
+### Empty_url_error
+
+**Error message**: URL was empty or null.
+
+This Error is thrown when no authority endpoint is provided in flows such as auth-code-with-certs.
```
Comment (tnorling) [inline lib/msal-node/docs/errors.md:125]: The library should default to `https://login.microsoftonline.com/common` if no authority is provided. This error should never be thrown unless there's a bug in the library.

File: lib/msal-node/docs/errors.md [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+        }
+    }
+```
+
+
+### Url_parse_error
+
+**Error message**: URL could not be parsed into appropriate segments. 
+
+This error is thrown when an invalid URL endpoint is provided as the authority value, in flows such as auth-code-with-certs.
```
Comment (tnorling) [inline lib/msal-node/docs/errors.md:145]: This is relevant to all flows
```suggestion
This error is thrown when an invalid URL is provided as the authority value.
```

File: lib/msal-node/docs/errors.md [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+❌ The following example will throw this error because `authority` endpoint provided is not a valid URL string:
+
+```javascript
+auth: {
+        clientId: "CLIENT_ID",
+        authority: "y",
```
Comment (tnorling) [inline lib/msal-node/docs/errors.md:152]: A more realistic example would be the URL without protocol:
```suggestion
        authority: "login.microsoftonline.com/common",
```

File: lib/msal-node/docs/errors.md [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+
+### Endpoints_resolution_error
+
+**Error Message**: Error: could not resolve endpoints. Please check network and try again.
+
+This error is thrown under the following scenarios:
+
+1. When a wrong authority endpoint is provided. Details: ClientAuthError: `openid_config_error`: Could not retrieve endpoints. Check your authority and verify the .well-known/openid-configuration endpoint returns the required endpoints. 
+
+Ensure that the authority provided matches the authority in your azure app registration.
```
Comment (tnorling) [inline lib/msal-node/docs/errors.md:35]:
```suggestion
Ensure that the authority provided matches the authority in your azure app registration or the `common`, `consumers` or `organizations` endpoints.
```

File: lib/msal-node/docs/errors.md [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+```javascript
+const config = {
+    auth: {
+        authority: "https://login.microsoftonline.com/<provide-authority-that-matches-your-app-registration>"
+        // ...   
+    }
+}
+```
+
+2. When there is no internet connectivity. Details: ClientConfigurationError: `untrusted_authority`: The provided authority is not a trusted authority. Please include this authority in the knownAuthorities config parameter. It is also thrown when no authority is appended to the authority endpoint.
```
Comment (tnorling) [inline lib/msal-node/docs/errors.md:45]: No internet connectivity is different from `untrusted_authority`, this should be 2 separate items

File: lib/msal-node/docs/errors.md [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+        }
+    }
+```
+
+
+### Request_cannot_be_made
+
+**Error message**: Token request cannot be made without authorization code or refresh token.
+
+This error is thrown when no authority is provided for auth flows such as auth-code-with-certs.
```
Comment (tnorling) [inline lib/msal-node/docs/errors.md:83]: This sounds like a bug if you're experiencing this when not providing authority. MSAL should be using the default.

File: lib/msal-node/docs/errors.md (hunk 3 above)
Comment (derisen) [inline lib/msal-node/docs/errors.md:125]:
```suggestion
This error is thrown when no authority endpoint is provided either in the configuration or in the token request objects.
```

File: lib/msal-node/docs/errors.md (hunk 4 above)
Comment (derisen) [inline lib/msal-node/docs/errors.md:145]:
```suggestion
This error is thrown when an invalid URL endpoint is provided as the authority value.
```

Let's not use sample names as flows (or maybe link the sample here?)

File: lib/msal-node/docs/errors.md [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+        clientId: "CLIENT_ID",
+        authority: "y",
+        clientCertificate: {
+            thumbprint: "YOUR_THUMBPRINT", 
+            privateKey: privateKey,
+        }
+    }
+```
+
+✔️ To resolve, ensure that the authority endpoint probided is valid.
```
Comment (derisen) [inline lib/msal-node/docs/errors.md:160]:
```suggestion
✔️ To resolve, ensure that the authority endpoint provided is a valid URL.
```

File: lib/msal-node/docs/errors.md [hunk 10]
```diff
@@ -0,0 +1,171 @@
+# Errors
```
Comment (jasonnutter) [inline lib/msal-node/docs/errors.md:1]:
```suggestion
# MSAL Node Errors
```

File: lib/msal-node/docs/errors.md [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+
```
Comment (jasonnutter) [inline lib/msal-node/docs/errors.md:2]: Can you add a short paragraph introduction explaining what is in this document?

File: lib/msal-node/docs/errors.md (hunk 7 above)
Comment (sameerag) [reply lib/msal-node/docs/errors.md:45]: +1. Node should detect network connectivity errors separately

**Review by sameerag — CHANGES_REQUESTED** [review]
Couple general comments:
* The format looks off, especially indentation of the index. 
* Can we make this tabular instead? This is mostly a reference page for all errors - a single table may be more readable than subheadings

**Pull request closed without merging**